scalar/scalar.go

```diff
@@ -208,7 +209,23 @@ func AppendToBuilder(bldr array.Builder, s Scalar) {
 		sb.Append(true)
 
 		v := s.(*Struct).Value
-		m := v.(map[string]any)
+		m, ok := v.(map[string]any)
+		if !ok {
+			reflectedMap := reflect.ValueOf(v)
+			if reflectedMap.Kind() != reflect.Map {
+				panic(fmt.Sprintf("expected map[string]any, got %T", v))
+			}
+
+			m = make(map[string]any, reflectedMap.Len())
+			for _, key := range reflectedMap.MapKeys() {
+				if key.Kind() != reflect.String {
+					panic(fmt.Sprintf("expected map[string]any, got %T", v))
+				}
+				value := reflectedMap.MapIndex(key)
+				m[key.String()] = value.Interface()
```

Comment:
(nit) although it's fine here, the good rule of thumb is to check via `can interface` method before calling `interface